src/aleph/http/client.clj

```diff
@@ -554,8 +554,11 @@
                             ;; might be different in case we use :multipart
                             (reset! save-body body))
 
-                          (netty/safe-execute ch
-                            (http/send-message ch true ssl? req' body))))
+                          (-> (netty/safe-execute ch
+                                                  (http/send-message ch true ssl? req' body))
+                              (d/catch' (fn [e]
+                                          (s/put! responses e)
```

Comment:
@arnaudgeiser 

@kachayev 's code in c9a6739db61948d64ef4c98e3df1d31753987fc3 wraps the throwable in an error-deferred, but this is raw. Which is correct?

Reply:
Both are correct actually and end up with the exact same result.
On my PR, I leverage the fact you can put a `Throwable` on `responses` which will be eventually coerced into
a manifold deferred : https://github.com/clj-commons/aleph/blob/master/src/aleph/http/client.clj#L581-L582

I did the minimum to solve the #542 issue. But I'm pretty sure we also need to wrap everything in a `try` like it's done here:
https://github.com/clj-commons/aleph/commit/c9a6739db61948d64ef4c98e3df1d31753987fc3#diff-f3c6a88143a34bab27cfd4dde73563c71de85a57da8133def509631b893d05ddR549-R550
I haven't done that modification since I haven't any test cases.

Reply:
Thanks for returning a deferred, I feel like that'll be safer/more future-proof.

What `try` is missing? You mean in the server-side code? Yeah, we probably need to add them to the stacks containing `send-message`

Reply:
`send-message` is used both on the client and on the server.
But I guess we are talking about the same thing though.

Reply:
Or audit the code to verify they're caught and handled already.